### ISSUE TITLE
Kafka sink should support Trace output

### ISSUE THREAD

**Issue by hvtuananh** [vectordotdev/vector#18252]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

The current Kafka sink only supports Log and Metric, even when using the `native` encoding.

### Configuration

```text
[sources.datadog_agent]
type = "datadog_agent"
address = "0.0.0.0:8126"
disable_logs = true
multiple_outputs = true

[sinks.kafka]
type = "kafka"
inputs = [ "datadog_agent.traces" ]
bootstrap_servers = "..."
topic = "..."
encoding.codec = "native"
```

### Version

vector 0.31.0 (x86_64-unknown-linux-gnu)

### Debug Output

```text
2023-08-15T17:47:22.402633Z ERROR vector::cli: Configuration error. error=Data type mismatch between datadog_agent.traces (Trace) and kafka (Log,Metric)
```

### Additional Context

As discussed in https://github.com/vectordotdev/vector/discussions/18177 this is just a matter of loosening the restriction of the current Kafka sink

**Comment by jszwedko** [comment]
Thanks @hvtuananh ! If I remember right you were planning to open a PR this. Happy to review when you do!

**Comment by hvtuananh** [comment]
@jszwedko yes, I created this issue because the Contributing guide stated that every change has an issue. As for the testing, I am not sure what we're looking for here, since the current Kafka sink config test is pretty barren. Could you elaborate on the test part?

**Comment by jszwedko** [comment]
Thanks @hvtuananh ! I left some feedback on the PR including where I think a test could be added.

**Comment by jszwedko** [comment]
@matejsp There is a PR for this: https://github.com/vectordotdev/vector/pull/18266. It's just awaiting the author responding to feedback. It's been a while though so it may be worth someone else picking that PR up if they are motivated.
